frontend/controller/actions/group.js

```diff
@@ -345,6 +348,80 @@ export default (sbp('sbp/selectors/register', {
       throw new GIErrorUIRuntimeError(L('Failed to leave group. {codeError}', { codeError: e.message }))
     }
   },
+  'gi.actions/group/autobanUser': async function (message: GIMessage, error: Object, attempt = 1) {
+    try {
+      // If we just joined, we're likely witnessing an old error that was handled
+      // by the existing members, so we shouldn't attempt to participate in voting
+      // in a proposal that has long since passed.
+      //
+      // NOTE: we cast to 'any' to work around flow errors
+      //       see: https://stackoverflow.com/a/41329247/1781435
+      const meta: Object = (message.opValue(): any).meta
+      const username = meta && meta.username
+      const groupID = message.contractID()
+      const contractState = sbp('state/vuex/state')[groupID]
+      const getters = sbp('state/vuex/getters')
+
+      if (username && getters.groupProfile(username)) {
+        console.warn(`autoBanSenderOfMessage: autobanning ${username} from ${groupID}`)
+        let proposal
+        let proposalHash
+        // find existing proposal if it exists
+        for (const hash in contractState.proposals) {
+          const prop = contractState.proposals[hash]
+          if (prop.status === STATUS_OPEN &&
+            prop.data.proposalType === PROPOSAL_REMOVE_MEMBER &&
+            prop.data.proposalData.member === username
+          ) {
+            proposal = prop
+            proposalHash = hash
+            break
+          }
+        }
```

Comment:
This could be made more concise, for example:
```suggestion
        // Find existing proposal if it exists.
        let [ proposalHash, proposal ] = Object.entries(contractState.proposals)
          .find(([hash, prop]) => (
            prop.status === STATUS_OPEN &&
            prop.data.proposalType === PROPOSAL_REMOVE_MEMBER &&
            prop.data.proposalData.member === username
          )) ?? []
```

Reply:
Yeah that looks like a good improvement, but is it me or would this code throw some kind of an error?

```
> d = { a: 4 }
{ a: 4 }
> Object.entries(d)
[ [ 'a', 4 ] ]
> let [i, j] = Object.entries(d).find(() => false)
Uncaught:
TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
```

Or am I doing something wrong?

Reply:
> let [i, j] = Object.entries(d).find(() => false)

Two variables `i` and `i` are set from the result of `find` which returns `undefined` always.
So it's just like `let [i, j] = undefined;`

If you try with the following code, it won't make any errors if object `d` is not empty.
`let [i, j] = Object.entries(d).find(() => true)`

Reply:
Ah, but that's the thing, `proposals` might not contain the proposal we're looking for, so in that case this would error.

Reply:
Sorry for leaving this case out.

Update: appended `?? []` to the suggested code to make it return and assign `undefined` values in case no proposal is found. But I find it a bit less clean now.